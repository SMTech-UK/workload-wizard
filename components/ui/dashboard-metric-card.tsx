import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-500">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-muted-foreground mt-1">{subtitle}</div>}
    </div>
  );
} 