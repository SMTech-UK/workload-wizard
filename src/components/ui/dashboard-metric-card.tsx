import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white dark:bg-zinc-900 shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-900 dark:text-white">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-gray-600 dark:text-gray-300 mt-1">{subtitle}</div>}
    </div>
  );
} 