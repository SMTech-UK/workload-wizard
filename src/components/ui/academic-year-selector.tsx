"use client"

import React from "react";
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
import { Badge } from "@/components/ui/badge";
import { useAcademicYear } from "@/hooks/useAcademicYear";
import { Calendar } from "lucide-react";

interface AcademicYearSelectorProps {
  className?: string;
  showActiveBadge?: boolean;
  onAcademicYearChange?: (academicYearId: string) => void;
}

export function AcademicYearSelector({ 
  className, 
  showActiveBadge = true,
  onAcademicYearChange 
}: AcademicYearSelectorProps) {
  const { 
    academicYears, 
    currentAcademicYear, 
    currentAcademicYearId, 
    changeAcademicYear 
  } = useAcademicYear();

  const handleAcademicYearChange = async (academicYearId: string) => {
    try {
      await changeAcademicYear(academicYearId);
      onAcademicYearChange?.(academicYearId);
    } catch (error) {
      console.error('Failed to change academic year:', error);
    }
  };

  // Function to format academic year name (e.g., "2025/26" -> "AY-25/26")
  const formatAcademicYearName = (yearName: string) => {
    // Extract the last 2 digits from each year part
    const parts = yearName.split('/');
    if (parts.length === 2) {
      const startYear = parts[0].slice(-2);
      const endYear = parts[1].slice(-2);
      return `AY-${startYear}/${endYear}`;
    }
    // Fallback to original name if format is unexpected
    return yearName;
  };

  if (!academicYears || academicYears.length === 0) {
    return (
      <div className={`flex items-center gap-2 text-muted-foreground ${className}`}>
        <Calendar className="h-4 w-4" />
        <span className="text-sm">No academic years configured</span>
      </div>
    );
  }

  return (
    <div className={`flex items-center gap-2 ${className}`}>
      <Select
        value={currentAcademicYearId || ""}
        onValueChange={handleAcademicYearChange}
      >
        <SelectTrigger className="w-auto min-w-[120px] flex items-center gap-2">
          <Calendar className="h-4 w-4 text-muted-foreground" />
          <span className="font-medium">
            {currentAcademicYear ? formatAcademicYearName(currentAcademicYear.name) : "Select AY"}
          </span>
        </SelectTrigger>
        <SelectContent>
          {academicYears.map((year: any) => (
            <SelectItem key={year._id} value={year._id}>
              <div className="flex items-center gap-2">
                <Calendar className="h-4 w-4 text-muted-foreground" />
                <span>{formatAcademicYearName(year.name)}</span>
                {year.isActive && showActiveBadge && (
                  <Badge variant="default" className="text-xs">Active</Badge>
                )}
                {year.isStaging && (
                  <Badge variant="secondary" className="text-xs">Staging</Badge>
                )}
              </div>
            </SelectItem>
          ))}
        </SelectContent>
      </Select>
    </div>
  );
} 