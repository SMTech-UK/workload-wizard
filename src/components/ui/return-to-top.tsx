"use client";

import { Button } from "@/components/ui/button";
import { ChevronUp, Home } from "lucide-react";
import { useState, useEffect } from "react";
import Link from "next/link";

export function ReturnToTop() {
  const [isVisible, setIsVisible] = useState(true); // Temporarily always visible for testing

  useEffect(() => {
    const toggleVisibility = () => {
      console.log("Scroll position:", window.pageYOffset);
      if (window.pageYOffset > 300) {
        console.log("Setting visible to true");
        setIsVisible(true);
      } else {
        console.log("Setting visible to false");
        setIsVisible(false);
      }
    };

    // Call once on mount to check initial state
    toggleVisibility();
    
    window.addEventListener("scroll", toggleVisibility);
    return () => window.removeEventListener("scroll", toggleVisibility);
  }, []);

  const scrollToTop = () => {
    console.log("Scrolling to top");
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  console.log("ReturnToTop component render, isVisible:", isVisible);

  return (
    <>
      {/* Return to Home button - top left */}
      <Link href="/">
        <Button
          size="sm"
          variant="outline"
          className="fixed top-4 left-4 z-50 shadow-lg hover:shadow-xl transition-all duration-200"
          aria-label="Return to home"
        >
          <Home className="h-4 w-4 mr-2" />
          Home
        </Button>
      </Link>

      {/* Return to Top button - bottom left */}
      <Button
        onClick={scrollToTop}
        size="sm"
        className="fixed bottom-4 left-4 z-50 rounded-full shadow-lg hover:shadow-xl transition-all duration-200 bg-blue-500 text-white"
        aria-label="Return to top"
      >
        <ChevronUp className="h-4 w-4" />
      </Button>
    </>
  );
} 