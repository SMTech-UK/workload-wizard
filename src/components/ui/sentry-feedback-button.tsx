"use client";

import { useEffect, useState } from "react";
import * as Sentry from "@sentry/nextjs";
import { Button } from "@/components/ui/button";
import { Bug, MessageCircle } from "lucide-react";
import { useModalAwareButton } from "@/hooks/useModalAwareButton";
import { createPortal } from "react-dom";

export function SentryFeedbackButton() {
  const [feedback, setFeedback] = useState<any>(null);
  const [widget, setWidget] = useState<any>(null);
  const [mounted, setMounted] = useState(false);
  const buttonRef = useModalAwareButton();

  useEffect(() => {
    setMounted(true);
    // Get the feedback instance on the client side
    setFeedback(Sentry.getFeedback());
  }, []);

  const handleFeedbackClick = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    
    if (!feedback) return;

    if (widget) {
      // If widget is open, close it
      widget.removeFromDom();
      setWidget(null);
    } else {
      // Create and show the form immediately
      feedback.createForm().then((newWidget: any) => {
        newWidget.appendToDom();
        newWidget.open();
        setWidget(newWidget);
      }).catch((error: any) => {
        console.error('Failed to create Sentry feedback form:', error);
      });
    }
  };

  if (!feedback || !mounted) return null;

  const button = (
    <Button
      ref={buttonRef}
      variant="destructive"
      size="lg"
      onClick={handleFeedbackClick}
      onMouseDown={(e) => {
        e.preventDefault();
        e.stopPropagation();
      }}
      className="fixed bottom-4 right-4 z-[99999] shadow-lg"
      style={{
        zIndex: 99999,
        pointerEvents: 'auto',
      }}
      aria-label="Report a bug or give feedback"
    >
      <Bug className="h-4 w-4 mr-2" />
      Report an Issue
    </Button>
  );

  // Render the button in a portal to ensure it's outside modal context
  return createPortal(button, document.body);
} 