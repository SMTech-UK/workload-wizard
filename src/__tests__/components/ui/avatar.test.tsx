import { render, screen } from '@testing-library/react';
import { Avatar, AvatarImage, AvatarFallback } from '@/components/ui/avatar';

// Custom query to find elements by data-slot attribute
const getByDataSlot = (container: HTMLElement, slotName: string) => {
  return container.querySelector(`[data-slot="${slotName}"]`);
};

const getAllByDataSlot = (container: HTMLElement, slotName: string) => {
  return container.querySelectorAll(`[data-slot="${slotName}"]`);
};

describe('Avatar Component - Comprehensive UI Testing', () => {
  describe('Basic Rendering and Functionality', () => {
    it('should render avatar with default styling', () => {
      // Arrange
      const { container } = render(<Avatar />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toBeInTheDocument();
      expect(avatar).toHaveClass('relative', 'flex', 'size-8', 'shrink-0', 'overflow-hidden', 'rounded-full');
    });

    it('should render avatar with custom className', () => {
      // Arrange
      const { container } = render(<Avatar className="custom-avatar" />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveClass('custom-avatar');
      expect(avatar).toHaveClass('relative', 'flex', 'size-8', 'shrink-0', 'overflow-hidden', 'rounded-full');
    });

    it('should render avatar with image', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarImage src="/test-image.jpg" alt="Test Avatar" />
        </Avatar>
      );

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      const image = getByDataSlot(container, 'avatar-image');
      expect(avatar).toBeInTheDocument();
      expect(image).toBeInTheDocument();
      expect(image).toHaveAttribute('src', '/test-image.jpg');
      expect(image).toHaveAttribute('alt', 'Test Avatar');
    });

    it('should render avatar with fallback', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarFallback>JD</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      const fallback = getByDataSlot(container, 'avatar-fallback');
      expect(avatar).toBeInTheDocument();
      expect(fallback).toBeInTheDocument();
      expect(fallback).toHaveTextContent('JD');
    });

    it('should render complete avatar with image and fallback', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarImage src="/test-image.jpg" alt="Test Avatar" />
          <AvatarFallback>JD</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      const image = getByDataSlot(container, 'avatar-image');
      const fallback = getByDataSlot(container, 'avatar-fallback');
      
      expect(avatar).toBeInTheDocument();
      expect(image).toBeInTheDocument();
      expect(fallback).toBeInTheDocument();
      expect(image).toHaveAttribute('src', '/test-image.jpg');
      expect(fallback).toHaveTextContent('JD');
    });
  });

  describe('Accessibility Testing', () => {
    it('should have proper semantic structure', () => {
      // Arrange
      const { container } = render(<Avatar />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar?.tagName).toBe('DIV');
      expect(avatar).toHaveAttribute('data-slot', 'avatar');
    });

    it('should support ARIA attributes', () => {
      // Arrange
      const { container } = render(<Avatar aria-label="User avatar" />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveAttribute('aria-label', 'User avatar');
    });

    it('should handle image accessibility properly', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarImage src="/test-image.jpg" alt="John Doe's profile picture" />
        </Avatar>
      );

      // Act & Assert
      const image = getByDataSlot(container, 'avatar-image');
      expect(image).toHaveAttribute('alt', 'John Doe\'s profile picture');
      expect(image).toHaveAttribute('src', '/test-image.jpg');
    });

    it('should support role attribute', () => {
      // Arrange
      const { container } = render(<Avatar role="img" aria-label="User avatar" />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveAttribute('role', 'img');
      expect(avatar).toHaveAttribute('aria-label', 'User avatar');
    });
  });

  describe('Responsive Behavior', () => {
    it('should maintain proper sizing on mobile screens', () => {
      // Arrange
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 375,
      });

      const { container } = render(<Avatar />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveClass('size-8');
      expect(avatar).toHaveClass('rounded-full');
    });

    it('should maintain proper sizing on tablet screens', () => {
      // Arrange
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 768,
      });

      const { container } = render(<Avatar />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveClass('size-8');
      expect(avatar).toHaveClass('rounded-full');
    });

    it('should maintain proper sizing on desktop screens', () => {
      // Arrange
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 1024,
      });

      const { container } = render(<Avatar />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveClass('size-8');
      expect(avatar).toHaveClass('rounded-full');
    });

    it('should handle custom sizing classes', () => {
      // Arrange
      const { container } = render(<Avatar className="size-12" />);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveClass('size-12');
      expect(avatar).toHaveClass('rounded-full');
    });
  });

  describe('Content Structure', () => {
    it('should handle image with various formats', () => {
      // Arrange
      const imageFormats = ['jpg', 'png', 'webp', 'svg'];
      
      imageFormats.forEach(format => {
        const { container } = render(
          <Avatar>
            <AvatarImage src={`/test-image.${format}`} alt={`Test ${format}`} />
          </Avatar>
        );

        // Act & Assert
        const image = getByDataSlot(container, 'avatar-image');
        expect(image).toHaveAttribute('src', `/test-image.${format}`);
        expect(image).toHaveAttribute('alt', `Test ${format}`);
      });
    });

    it('should handle fallback with initials', () => {
      // Arrange
      const initials = ['JD', 'AB', 'XYZ', '123'];
      
      initials.forEach(initial => {
        const { container } = render(
          <Avatar>
            <AvatarFallback>{initial}</AvatarFallback>
          </Avatar>
        );

        // Act & Assert
        const fallback = getByDataSlot(container, 'avatar-fallback');
        expect(fallback).toHaveTextContent(initial);
      });
    });

    it('should handle fallback with emoji', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarFallback>ðŸ‘¤</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      const fallback = getByDataSlot(container, 'avatar-fallback');
      expect(fallback).toHaveTextContent('ðŸ‘¤');
    });

    it('should handle empty fallback', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarFallback></AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      const fallback = getByDataSlot(container, 'avatar-fallback');
      expect(fallback).toBeInTheDocument();
      expect(fallback).toHaveClass('bg-muted', 'flex', 'size-full', 'items-center', 'justify-center', 'rounded-full');
    });
  });

  describe('Edge Cases and Error Handling', () => {
    it('should handle very long fallback text', () => {
      // Arrange
      const longText = 'A'.repeat(100);
      const { container } = render(
        <Avatar>
          <AvatarFallback>{longText}</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      const fallback = getByDataSlot(container, 'avatar-fallback');
      expect(fallback).toHaveTextContent(longText);
      expect(fallback).toHaveClass('bg-muted', 'flex', 'size-full', 'items-center', 'justify-center', 'rounded-full');
    });

    it('should handle special characters in fallback', () => {
      // Arrange
      const specialChars = 'User with special chars: <>&"\'<>';
      const { container } = render(
        <Avatar>
          <AvatarFallback>{specialChars}</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      const fallback = getByDataSlot(container, 'avatar-fallback');
      expect(fallback).toHaveTextContent(specialChars);
    });

    it('should handle multiple avatars', () => {
      // Arrange
      const { container } = render(
        <div>
          <Avatar>
            <AvatarFallback>JD</AvatarFallback>
          </Avatar>
          <Avatar>
            <AvatarFallback>AB</AvatarFallback>
          </Avatar>
          <Avatar>
            <AvatarFallback>CD</AvatarFallback>
          </Avatar>
        </div>
      );

      // Act & Assert
      const avatars = getAllByDataSlot(container, 'avatar');
      const fallbacks = getAllByDataSlot(container, 'avatar-fallback');
      expect(avatars).toHaveLength(3);
      expect(fallbacks).toHaveLength(3);
      expect(fallbacks[0]).toHaveTextContent('JD');
      expect(fallbacks[1]).toHaveTextContent('AB');
      expect(fallbacks[2]).toHaveTextContent('CD');
    });

    it('should handle broken image gracefully', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarImage src="/broken-image.jpg" alt="Broken image" />
          <AvatarFallback>JD</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      const image = getByDataSlot(container, 'avatar-image');
      const fallback = getByDataSlot(container, 'avatar-fallback');
      expect(image).toHaveAttribute('src', '/broken-image.jpg');
      expect(fallback).toHaveTextContent('JD');
    });
  });

  describe('Snapshot Testing for UI Consistency', () => {
    it('should match snapshot for default avatar', () => {
      // Arrange
      const { container } = render(<Avatar />);

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for avatar with image', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarImage src="/test-image.jpg" alt="Test Avatar" />
        </Avatar>
      );

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for avatar with fallback', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarFallback>JD</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for complete avatar', () => {
      // Arrange
      const { container } = render(
        <Avatar>
          <AvatarImage src="/test-image.jpg" alt="Test Avatar" />
          <AvatarFallback>JD</AvatarFallback>
        </Avatar>
      );

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for multiple avatars', () => {
      // Arrange
      const { container } = render(
        <div>
          <Avatar>
            <AvatarFallback>JD</AvatarFallback>
          </Avatar>
          <Avatar>
            <AvatarFallback>AB</AvatarFallback>
          </Avatar>
        </div>
      );

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Performance Testing', () => {
    it('should handle rapid re-renders efficiently', () => {
      // Arrange
      const { rerender } = render(<Avatar>Initial Avatar</Avatar>);

      // Act & Assert
      expect(screen.getByText('Initial Avatar')).toBeInTheDocument();

      // Rerender with different content
      rerender(<Avatar>Updated Avatar</Avatar>);

      expect(screen.getByText('Updated Avatar')).toBeInTheDocument();
      expect(screen.queryByText('Initial Avatar')).not.toBeInTheDocument();
    });

    it('should handle className changes efficiently', () => {
      // Arrange
      const { rerender, container } = render(<Avatar className="size-8">Small</Avatar>);

      // Act & Assert
      const avatar = getByDataSlot(container, 'avatar');
      expect(avatar).toHaveClass('size-8');

      // Change className
      rerender(<Avatar className="size-12">Large</Avatar>);

      const updatedAvatar = getByDataSlot(container, 'avatar');
      expect(updatedAvatar).toHaveClass('size-12');
      expect(updatedAvatar).not.toHaveClass('size-8');
    });

    it('should handle multiple avatar renders efficiently', () => {
      // Arrange
      const avatars = Array.from({ length: 100 }, (_, i) => `Avatar ${i + 1}`);

      const startTime = performance.now();

      // Act
      render(
        <div>
          {avatars.map((text, index) => (
            <Avatar key={index}>
              <AvatarFallback>{text}</AvatarFallback>
            </Avatar>
          ))}
        </div>
      );

      const endTime = performance.now();

      // Assert
      expect(screen.getByText('Avatar 1')).toBeInTheDocument();
      expect(screen.getByText('Avatar 100')).toBeInTheDocument();
      expect(endTime - startTime).toBeLessThan(1000); // Should render within 1 second
    });
  });
}); 