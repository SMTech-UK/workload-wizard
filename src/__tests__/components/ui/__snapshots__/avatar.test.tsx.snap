// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for avatar with fallback 1`] = `
<span
  class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
  data-slot="avatar"
>
  <span
    class="bg-muted flex size-full items-center justify-center rounded-full"
    data-slot="avatar-fallback"
  >
    JD
  </span>
</span>
`;

exports[`Avatar Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for avatar with image 1`] = `
<span
  class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
  data-slot="avatar"
/>
`;

exports[`Avatar Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for complete avatar 1`] = `
<span
  class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
  data-slot="avatar"
>
  <span
    class="bg-muted flex size-full items-center justify-center rounded-full"
    data-slot="avatar-fallback"
  >
    JD
  </span>
</span>
`;

exports[`Avatar Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for default avatar 1`] = `
<span
  class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
  data-slot="avatar"
/>
`;

exports[`Avatar Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for multiple avatars 1`] = `
<div>
  <span
    class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
    data-slot="avatar"
  >
    <span
      class="bg-muted flex size-full items-center justify-center rounded-full"
      data-slot="avatar-fallback"
    >
      JD
    </span>
  </span>
  <span
    class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
    data-slot="avatar"
  >
    <span
      class="bg-muted flex size-full items-center justify-center rounded-full"
      data-slot="avatar-fallback"
    >
      AB
    </span>
  </span>
</div>
`;
